src/main/java/org/jenkinsci/maven/plugins/hpi/TestDependencyMojo.java

```diff
@@ -550,6 +549,12 @@ private static Map<String, String> scanWar(File war, MavenSession session, Maven
                         if (version == null) {
                             throw new IllegalArgumentException("Failed to determine version for " + name);
                         }
+                        // handle any extra info in snapshots e.g. " (private-abcd1234-username)"
+                        version = version.replaceFirst(" [(].+[)]$", "");
+                        if (version.endsWith("-SNAPSHOT")) {
+                            getLog().warn("War contains a SNAPSHOT of " + groupId + ":" + artifactId
+                                    + " build will not be fully repeatable");
```

Comment:
WAR is capitalized incorrectly, and there is a run-on sentence without punctuation:

```suggestion
                            getLog().warn("WAR contains a SNAPSHOT of " + groupId + ":" + artifactId
                                    + "; build will not be fully repeatable");
```